src/frontend/EdFi.DataManagementService.Frontend.AspNetCore/Program.cs

```diff
@@ -51,15 +55,23 @@ void InitializeDatabase(WebApplication app)
 {
     if (app.Services.GetRequiredService<IOptions<AppSettings>>().Value.DeployDatabaseOnStartup)
     {
-        var result = app
-            .Services.GetRequiredService<IDatabaseDeploy>()
-            .DeployDatabase(
-                app.Services.GetRequiredService<IOptions<ConnectionStrings>>().Value.DatabaseConnection
-            );
-        if (result is DatabaseDeployResult.DatabaseDeployFailure failure)
+        try
+        {
+            var result = app
+                .Services.GetRequiredService<IDatabaseDeploy>()
+                .DeployDatabase(
+                    app.Services.GetRequiredService<IOptions<ConnectionStrings>>().Value.DatabaseConnection
+                );
+            if (result is DatabaseDeployResult.DatabaseDeployFailure failure)
+            {
+                app.Logger.LogCritical(failure.Error, "Database Deploy Failure");
+                Environment.Exit(-1);
+            }
+        }
+        catch (Exception ex)
         {
-            app.Logger.LogCritical("Database Deploy Failure");
-            throw failure.Error;
+            app.Logger.LogCritical(ex, "Database Deploy Failure");
+            Environment.Exit(-1);
```

Comment:
Previously if an error occurred - for example, password authentication failure - this threw an unhandled exception and the application hung until Control-C.